Orange/widgets/evaluate/owpredictions.py

```diff
@@ -230,20 +230,22 @@ def set_predictor(self, predictor=None, id=None):
                 PredictorSlot(predictor, predictor.name, None)
 
     def _set_class_var(self):
+        self.class_var = self.data and self.data.domain.class_var
```

Comment:
`self.class_var` now uses the variable from test data instead of the one from models.
This is wrong, test data's target is mostly irrelevant (we also allow it to be None) and should only be used to double check that the user is not making a mistake by using incompatible models and data.

Arguably, the mismatch error could have been a warning instead, since if we allow no target, why should we stop everything when it is present, but different. This was probably your thinking as well, when you removed `self.class_var = None` after the mismatch error? I think the policy in Orange is to stop everything on Errors and use Warnings when something is not as it should be, but the work can continue.

Reply:
An example of this causing problems is the same as the one I mentioned above.
> File (iris) -> Logistic Regression -> Predictions <- File (iris12),
> where iris12 is a file with only 2 class values from iris, e.g. first 100 instances
The control area changes to show only 2 class values after test data is connected. There are also only 2 probabilities shown, although the model should report 3. Compare this with removing the target from test data.